stl/inc/xstring

```diff
@@ -2854,17 +3059,38 @@ public:
             _Result._Res = _BUF_SIZE + 1;
         }
         _My_data._Orphan_all();
+        _ASAN_STRING_CLEAR(*this);
         _Tidy_init();
         return _Result;
     }
 #endif // _HAS_CXX20
 
     _CONSTEXPR20 basic_string& operator=(basic_string&& _Right) noexcept(
-        noexcept(_Move_assign(_Right, _Choose_pocma<_Alty>{}))) {
-        if (this != _STD addressof(_Right)) {
-            _Move_assign(_Right, _Choose_pocma<_Alty>{});
+        !is_same_v<_Choose_pocma<_Alty>, _No_propagate_allocators>) {
+        if (this == _STD addressof(_Right)) {
+            return *this;
         }
 
+        auto& _Al       = _Getal();
+        auto& _Right_al = _Right._Getal();
+        using _POCMA    = _Choose_pocma<_Alty>;
+        if constexpr (is_same_v<_POCMA, _Propagate_allocators>) {
+            if (_Al != _Right_al) {
+                // intentionally slams into noexcept on OOM, TRANSITION, VSO-466800
+                _Mypair._Myval2._Orphan_all();
+                _Mypair._Myval2._Reload_proxy(_GET_PROXY_ALLOCATOR(_Alty, _Al), _GET_PROXY_ALLOCATOR(_Alty, _Right_al));
+            }
+
+        } else if constexpr (is_same_v<_POCMA, _No_propagate_allocators>) {

```

Comment:
```suggestion
        } else {

```

Reply:
There is still the  `_Equal_allocators` case, so that is not possible